luigi/contrib/k8s_job.py

```diff
@@ -126,46 +138,63 @@ def spec_schema(self):
         For more informations please refer to:
         http://kubernetes.io/docs/user-guide/pods/multi-container/#the-spec-schema
         """
+        raise NotImplementedError("subclass must define spec_schema")
 
     @property
     def max_retrials(self):
         """
         Maximum number of retrials in case of failure.
         """
-        return 0
+        return configuration.get_config().get("k8s", "max_retrials", 0)
 
     def __track_job(self):
         """Poll job status while active"""
         while (self.__get_job_status() == "running"):
-            self.__logger.debug("Kubernetes job " + self.uu_name + " is still running")
+            self.__logger.debug("Kubernetes job " + self.uu_name
+                                + " is still running")
             time.sleep(self.__POLL_TIME)
         if(self.__get_job_status() == "succeeded"):
             self.__logger.info("Kubernetes job " + self.uu_name + " succeeded")
+            # Use signal_complete to notify of job completion
+            self.signal_complete()
         else:
             raise RuntimeError("Kubernetes job " + self.uu_name + " failed")
 
+    def signal_complete(self):
```

Comment:
could you make this function optional? So if I have a distributed FS and a Job the writes the output file, I can avoid to implement this.